.circleci/scripts/find_changed_packages.js

```diff
@@ -38,101 +30,17 @@ const getWorkspacesInfo = () => {
   const e2eDirHasTestFiles = e2eDirExists && readdirSync(e2eDir).some(file => file.endsWith('.test.ts'))
   console.log('e2eDir:', e2eDir, 'exists:', e2eDirExists, 'hasTestFiles:', e2eDirHasTestFiles)
   return e2eDirHasTestFiles
 }
 
-const findChangedPackages = (userInputBaseCommit, workspaceInfo) => {
-  const changedFiles = getChangedFiles(userInputBaseCommit)
-  console.log('Changed files:', changedFiles)
-
-  const changedPackageLocation = new Set(changedFiles.map(path => path.split('/').slice(0, 2).join('/')))
-
-  const changedPackages = Array.from(changedPackageLocation).map(packageDir =>
-    Object.keys(workspaceInfo).find(key => workspaceInfo[key].location === packageDir),
-  )
-  const hasChangedFilesOutsidePackage = changedPackages.some(pkg => !pkg)
-
-  console.log('Changed packages:', changedPackages)
-  console.log('Has changes outside packages directory:', hasChangedFilesOutsidePackage)
-  return { changedPackages, hasChangedFilesOutsidePackage }
-}
-
-const getDependenciesFromChangedPackages = (changedPackages, workspaceInfo) => {
-  const dependencyMapping = generateDependencyMapping(workspaceInfo)
-  console.log('Dependency mapping:', dependencyMapping)
-
-  const changedPackagesDependencies = new Set(
-    changedPackages.flatMap(package => dependencyMapping[package]).filter(Boolean),
-  )
-  console.log('Changed packages dependencies:', changedPackagesDependencies)
-  const dependenciesLocation = Array.from(changedPackagesDependencies)
-    .map(pkg => workspaceInfo[pkg].location)
-    .sort()
-  console.log('Changed packages with dependencies location:', dependenciesLocation)
-
-  const packagesToTest = dependenciesLocation.map(location => location.replace('packages/', ''))
-  return packagesToTest
-}
-
-const getPackagesToTest = (userInputBaseCommit, workspaceInfo, allPackages) => {
-  const { changedPackages, hasChangedFilesOutsidePackage } = findChangedPackages(userInputBaseCommit, workspaceInfo)
-  return hasChangedFilesOutsidePackage
-    ? allPackages
-    : getDependenciesFromChangedPackages(changedPackages, workspaceInfo)
-}
-
 const main = () => {
-  const args = process.argv.slice(2)
-  const userInputBaseCommit = args.length > 0 ? args[0] : null
-  if (userInputBaseCommit) {
-    console.log(`User provided base commit: ${userInputBaseCommit}`)
-  }
-
   const workspaceInfo = getWorkspacesInfo()
-  const allPackages = Object.keys(workspaceInfo)
-    .map(pkg => workspaceInfo[pkg].location.replace('packages/', ''))
-    .sort()
-  // on main branch, we want to test all packages
-  const packagesToTest =
-    process.env.CIRCLE_BRANCH === 'main'
-      ? allPackages
-      : getPackagesToTest(userInputBaseCommit, workspaceInfo, allPackages)
-  console.log('Packages to test:', packagesToTest)
-
-  const e2ePackagesToTest = packagesToTest.filter(hasE2eTests)
+  const e2ePackagesToTest = Object.keys(Object.fromEntries(Object.entries(workspaceInfo).filter(([name, details]) => hasE2eTests(details.location))))
```

Comment:
This seems over complicated. Why not:
```suggestion
  const e2ePackagesToTest = Object.entries(workspaceInfo).filter(([_name, details]) => hasE2eTests(details.location)).map(([name, _details]) => name)
```